Sources/Recycling/ListViews/Wishlist/ListView/WishListView.swift

```diff
@@ -0,0 +1,134 @@
+    func wishListView(_ wishListView: WishListView, modelAtIndex index: Int) -> WishListViewModel
+    func wishListViewCell(_ wishlistListViewCell: WishListViewCell, loadImageForModel model: WishListViewModel, imageWidth: CGFloat, completion: @escaping ((UIImage?) -> Void))
+    func wishListViewCell(_ wishlistListViewCell: WishListViewCell, cancelLoadingImageForModel model: WishListViewModel, imageWidth: CGFloat)
+}
+
+public class WishListView: UIView {
+    public static let estimatedRowHeight: CGFloat = 300
+    
+    // MARK: - Internal properties
+    
+    private lazy var tableView: UITableView = {
+        let tableView = UITableView(frame: .zero, style: .plain)
+        tableView.translatesAutoresizingMaskIntoConstraints = false
+        tableView.delegate = self
+        tableView.dataSource = self
+        tableView.backgroundColor = .milk
+        tableView.rowHeight = WishListView.estimatedRowHeight
+        tableView.estimatedRowHeight = WishListView.estimatedRowHeight
+        tableView.separatorStyle = .none
+        return tableView
+    }()
+    
+    // MARK: - Public properties
+    
+    public weak var delegate: WishListViewDelegate?
+    public weak var dataSource: WishListViewDataSource?
+    
+    // MARK: - Setup
+    
+    public init(delegate: WishListViewDelegate, dataSource: WishListViewDataSource) {
+        super.init(frame: .zero)
+        
+        self.delegate = delegate
+        self.dataSource = dataSource
+        
+        setupView()
+    }
+    
+    public override init(frame: CGRect) {
+        super.init(frame: frame)
+        setupView()
+    }
+    
+    public required init?(coder aDecoder: NSCoder) {
+        super.init(coder: aDecoder)
+        setupView()
+    }
+    
+    private func setupView() {
+        tableView.register(WishListViewCell.self)
+        addSubview(tableView)
+        tableView.fillInSuperview()
+    }
+    
+    // MARK: - Public
+    
+    public func reloadData() {
+        tableView.reloadData()
+    }
+    
+    public func scrollToTop(animated: Bool = true) {
+        tableView.scrollRectToVisible(CGRect(x: 0, y: 0, width: 1, height: 1), animated: animated)
+    }
+}
+
+// MARK: - UITableViewDataSource
+
+extension WishListView: UITableViewDataSource {
+    public func tableView(_ tableView: UITableView, numberOfRowsInSection section: Int) -> Int {
+        return dataSource?.numberOfItems(inWishListView: self) ?? 0
+    }
+    
+    public func tableView(_ tableView: UITableView, cellForRowAt indexPath: IndexPath) -> UITableViewCell {
+        let cell = tableView.dequeue(WishListViewCell.self, for: indexPath)
+        
+        let colors: [UIColor] = [.toothPaste, .mint, .banana, .salmon]
+        let color = colors[indexPath.row % 4]
+        
+        cell.loadingColor = color
+        cell.dataSource = self
+        
+        if let model = dataSource?.wishListView(self, modelAtIndex: indexPath.row) {
+            cell.model = model
```

Comment:
This could have been:

```swift
cell.model = dataSource?.wishListView(self, modelAtIndex: indexPath.row)
```

Not sure if it was needed to unwrap the model.